service/qos/service_level_controller.hh

```diff
@@ -222,7 +222,7 @@ private:
      */
     service_level& get_service_level(sstring service_level_name) {
         auto sl_it = _service_levels_db.find(service_level_name);
-        if (sl_it == _service_levels_db.end() && sl_it->second.marked_for_deletion) {
+        if (sl_it == _service_levels_db.end() || sl_it->second.marked_for_deletion) {
```

Comment:
Please fold into the commit that introduced the problem.

Reply:
I did it that way to match the fix in enterprise (I generally tried to not touch the original series too much), but I can fold in here, no problem